plugins/filter/crc32.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2022, Julien Riou <julien@riou.xyz>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.errors import AnsibleFilterError
+from ansible.module_utils.six import string_types
+from ansible.module_utils.common.text.converters import to_bytes
+
+try:
+    from zlib import crc32
+    HAS_ZLIB = True
+except ImportError:
+    HAS_ZLIB = False
+
+
+DOCUMENTATION = '''
+  name: crc32
+  short_description: Generate a CRC32 checksum
+  version_added: 5.4.0
+  description:
+    - Checksum a string using CRC32 algorithm and return its hexadecimal representation.
+  options:
+    _input:
+      description:
+        - The string to checksum.
+      type: string
+      required: true
+  author:
+    - Julien Riou
+'''
+
+EXAMPLES = '''
+  - name: Checksum a test string
+    ansible.builtin.debug:
+      msg: "{{ 'test' | community.general.crc32 }}"
+'''
+
+RETURN = '''
+  _value:
+    description: CRC32 checksum.
+    type: string
+'''
+
+
+def crc32s(value):
+    if not isinstance(value, string_types):
```

Comment:
You might want to use `ansible.module_utils.common.colections.is_string()`.

Reply:
Replaced by is_string. Thank you for the suggestion!